prow/tide/tide.go

```diff
@@ -493,26 +493,49 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 
 // isPassingTests returns whether or not all contexts set on the PR except for
 // the tide pool context are passing.
-func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest) bool {
+func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest, bp config.TideBranchProtection) bool {
 	log = log.WithFields(pr.logFields())
 	contexts, err := headContexts(log, ghc, &pr)
 	if err != nil {
 		log.WithError(err).Error("Getting head commit status contexts.")
 		// If we can't get the status of the commit, assume that it is failing.
 		return false
 	}
+
+	requiredChecksSet := sets.NewString()
+	if bp.UseRequiredCheckOnly {
+		requiredChecks, err := ghc.GetRequiredChecks(
+			string(pr.Repository.Owner.Login),
+			string(pr.Repository.Name),
+			string(pr.BaseRef.Name))
+		if err != nil {
+			log.WithError(err).Errorf("getting required checks")
+			return false
+		}
+		requiredChecksSet.Insert(requiredChecks...)
+	}
+	combinedSet := sets.NewString()
```

Comment:
I think this variable name is a misnomer. It is the set of contexts seen on the PR, nothing is combined?